internal/addon/coo.go

```diff
@@ -0,0 +1,53 @@
+package addon
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func InstallCOO(ctx context.Context, k8s client.Client, logger logr.Logger, isHub bool, opts Options) (bool, error) {
+	if !cooDependantEnabled(opts) {
+		return false, nil
+	}
+
+	// Currently, the InstallCOO option is only relevant for hub clusters
+	// since we don't have k8s clients for the spokes
+	if !isHub {
+		return true, nil
+	}
+
+	cooSub := &operatorv1alpha1.Subscription{}
+	key := client.ObjectKey{Name: cooSubscriptionName, Namespace: cooSubscriptionNamespace}
+	if err := k8s.Get(ctx, key, cooSub, &client.GetOptions{}); err != nil && !k8serrors.IsNotFound(err) {
```

Comment:
I guess this also requires permissions update to get/read the subscription or not?

Reply:
https://github.com/stolostron/multicluster-observability-operator/pull/1982